### PR TITLE
execute: pass ambient caps from PAM through to invoked service

### PR DESCRIPTION
**Pull request by poettering**
If a PAM service sets some ambient caps, we should honour that, hence query it, and merge it with our own ambient settings.

This needs to be done manually since otherwise dropping privs via setresuid() will undo all such caps, and we need to manually tweak things to keep them.

### CODE REVIEW

**Comment by poettering** [comment]
This is split out of #26548. This definitely makes sense on its own, as it fixes hooking any PAM module that sets ambient caps into a systemd service via PAMName=.

#26548 contains a test that verifies this works, once pam_systemd itself becomes one of the services that can set ambient caps. But given how relatively straightforward this is, I think we can merge that independently and before that later test.

File: src/core/execute.c [hunk 1]
```diff
@@ -4273,6 +4276,7 @@ static int exec_child(
         int secure_bits;
         _cleanup_free_ gid_t *gids_after_pam = NULL;
         int ngids_after_pam = 0;
+        uint64_t capability_ambient_set = UINT64_MAX;
```
Comment (yuwata) [inline src/core/execute.c]: The initialization seems not necessary, right?

**Review by yuwata — APPROVED** [review]
LGTM. One minor comment.

File: src/core/execute.c (hunk 1 above)
Comment (poettering) [reply src/core/execute.c]: true. fixed.

**Review by keszybz — COMMENTED** [review]
Looks reasonable.

**Pull request merged**